app/scripts/background.js

```diff
@@ -2,9 +2,22 @@
  * @file The entry point for the web extension singleton process.
  */
 
+// disabled to allow importing initial state hooks first
+/* eslint-disable import/order,import/first */
+
+import ExtensionPlatform from './platforms/extension';
+import LocalStore from './lib/local-store';
+import ReadOnlyNetworkStore from './lib/network-store';
+
+const inTest = process.env.IN_TEST;
```

Comment:
Do these declarations have to happen before `import { setupInitialStateHooks } from './lib/setup-initial-state-hooks';`? Why is that?

Reply:
`platform` and `localStore` are both passed into `setupInitialStateHooks`. The intent here was for that call to happen as early as possible, so that an error later in this file would include the state snapshot in the error report

Reply:
Oh I see, the comment is misleading at the moment. I'll update it.

Reply:
Okay, that makes sense.

There is another thing I still don't understand though.  Is there a functional difference between the code as it is:
```
import ExtensionPlatform from './platforms/extension';
import LocalStore from './lib/local-store';
import ReadOnlyNetworkStore from './lib/network-store';

const inTest = process.env.IN_TEST;
const localStore = inTest ? new ReadOnlyNetworkStore() : new LocalStore();
const platform = new ExtensionPlatform();

import { setupInitialStateHooks } from './lib/setup-initial-state-hooks';

setupInitialStateHooks({ localStore, platform });
```
And this alternative order of lines:
```
import ExtensionPlatform from './platforms/extension';
import LocalStore from './lib/local-store';
import ReadOnlyNetworkStore from './lib/network-store';
import { setupInitialStateHooks } from './lib/setup-initial-state-hooks';

const inTest = process.env.IN_TEST;
const localStore = inTest ? new ReadOnlyNetworkStore() : new LocalStore();
const platform = new ExtensionPlatform();

setupInitialStateHooks({ localStore, platform });
```

?

Reply:
Nope. I can re-order them to put imports first. The goal here was just to get this function call to happen before the imports, so that any errors on import would also be caught.

Though.... I just remembered that imports are hoisted. So this wouldn't even have that effect. Hmm...

Reply:
I revised this significantly here: https://github.com/MetaMask/metamask-extension/pull/20491/commits/db52b4e1d2fc590018e1905812bf7c6743923f52